MCFM-JHUGen/src/User/mdata.f

```diff
@@ -349,10 +350,34 @@ block data electroweak_input
 c     anomalous couplings for triple and quartic gauge boson coupling with d_4 (SM: set all to 0)
       data dFour_A / (0d0,0d0) /
       data dFour_Z / (0d0,0d0) /
+      
+c     right handed Z couplings to charged leptons (set separately for each Z)
+      data reZ / 0.54837d0 / ! SM = (-2*-1*xw_inp)/(2*sqrt(xw_inp*(1-xw_inp)))
 
+c     left handed Z couplings to charged leptons (set separately for each Z)
+      data leZ / -0.63760d0 / ! SM = (-1-2*-1*xw_inp)/(2*sqrt(xw_inp*(1-xw_inp)))
 
-c     End anomalous couplings
+c     left handed Z couplings to neutrinos 
+      data lnZ / 1.18598d0 / ! SM = (1)/(2*sqrt(xw_inp*(1-xw_inp))
+
+c     right handed Z couplings to neutrinos
+      data rnZ / 0d0 / ! SM = 0 
+
+c     Anomalous Couplings for the Z to up type quarks
+c     Note, these are called as shifts to SM values
 
+      data clanou / 0d0 / ! SM = 0
+      data cranou / 0d0 / ! SM = 0
+
+c     Anomalous Couplings for the Z to down type quarks
+c     Note, these are called as shifts to SM values
+      
+      data clanod / 0d0 / ! SM = 0
+      data cranod / 0d0 / ! SM = 0
```

Comment:
Hardcoding these couplings is a very bad idea. They should be predetermined by the time ewscheme is used and couplings are set.

Reply:
No longer hard coded. Now, they are set to 0 by default. It is up to the user to ensure they are set properly if they decide to use anomalous Zff couplings.